lib/gringotts/gateways/trexle.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Gringotts.Gateways.Trexle do
+  
+  @moduledoc """
+  Functions supported for working with Trexle payment gateway:
+  *
+  *
+  """
+  @base_url "https://core.trexle.com/api/v1/"
+
+  alias Gringotts.Gateways.Trexle.ResponseHandler, as: ResponseParser
+  use Gringotts.Gateways.Base
+  use Gringotts.Adapter, required_config: [:api_key, :default_currency]
+
+  @spec authorize(Float, Map, List) :: Map
+  def authorize(amount, payment, opts \\ []) do
+    params = create_params_for_auth_or_purchase(amount, payment, opts, false)
+    commit(:post, "charges", params, opts)
+  end
+
+  @spec purchase(Float, Map, List) :: Map
+  def purchase(amount, payment, opts \\ []) do
+    params = create_params_for_auth_or_purchase(amount, payment, opts)
+    commit(:post, "charges", params, opts)
+  end
+
+  @spec capture(String.t, Float, List) :: Map
+  def capture(charge_token, amount, opts \\ []) do
+    params = [amount: amount]
+    commit(:put, "charges/#{charge_token}/capture", params, opts)
+  end
+  
+  @spec refund(Float, String.t, List) :: Map 
+  def refund(amount, charge_token, opts \\ []) do
```

Comment:
@jyotigautam signature for this method is `amount, charge_token, opts` for capture its `capture_token, amount, opts` why this disparity ?

Reply:
make sure this convention is consistent all across the file.

Reply:
@jyotigautam did you correct this? why is my comment outdated when you have haven't worked on this one?